src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -224,6 +224,12 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
                         namedTypeSymbol.GetMembers().OfType<IFieldSymbol>().Where(m => !IsSerializable(m.Type));
                     foreach (IFieldSymbol field in nonSerializableFields)
                     {
+                        // Only process instance fields
+                        if (field.IsConst || field.IsStatic)
```

Comment:
❔ Does `IsConst` not imply `IsStatic`? In other words, is it sufficient to use the following:

```csharp
if (field.IsStatic)
{
  continue;
}
```

Reply:
Yep, changed.